packages/flutter_tools/lib/src/commands/migrate_resolve_conflicts.dart

```diff
@@ -0,0 +1,309 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../migrate/migrate_manifest.dart';
+import '../migrate/migrate_utils.dart';
+import '../project.dart';
+import '../runner/flutter_command.dart';
+import 'migrate.dart';
+
+/// Flutter migrate subcommand that guides the developer through conflicts,
+/// allowing them to accept the original, the new lines, or skip and resolve manually.
+class MigrateResolveConflictsCommand extends FlutterCommand {
+  MigrateResolveConflictsCommand({
+    required this.logger,
+    required this.fileSystem,
+    required this.terminal,
+  }) {
+    requiresPubspecYaml();
+    argParser.addOption(
+      'working-directory',
+      help: 'Specifies the custom migration working directory used to stage and edit proposed changes. '
+            'This path can be absolute or relative to the flutter project root.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'project-directory',
+      help: 'The root directory of the flutter project.',
+      valueHelp: 'path',
+    );
+    argParser.addOption(
+      'context-lines',
+      defaultsTo: '5',
+      help: 'The number of lines of context to show around the each conflict. Defaults to 5.',
+    );
+    argParser.addFlag(
+      'confirm-commit',
+      defaultsTo: true,
+      help: 'Indicates if proposed changes require user verification before writing to disk.',
+    );
+  }
+
+  final Logger logger;
+
+  final FileSystem fileSystem;
+
+  final Terminal terminal;
+
+  @override
+  final String name = 'resolve-conflicts';
+
+  @override
+  final String description = 'Prints the current status of the in progress migration.';
+
+  @override
+  String get category => FlutterCommandCategory.project;
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => const <DevelopmentArtifact>{};
+
+  static const String _conflictStartMarker = '<<<<<<<';
+  static const String _conflictDividerMarker = '=======';
+  static const String _conflictEndMarker = '>>>>>>>';
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
```

Comment:
This method is over 200 lines long. This makes it difficult to review, and likely also makes it difficult to unit test. I don't have any concrete suggestions for how to split it up, but a guiding principle should be splitting it up in such a way that it is straightforward to write tests.

Reply:
Refactored!